front/lib/actions/mcp_internal_actions/servers/run_agent/index.ts

```diff
@@ -258,13 +259,21 @@ export default async function createServer(
           fileOrContentFragmentIds,
           conversationId,
         },
-        { sendNotification, _meta }
+        { sendNotification, _meta, signal }
       ) => {
         assert(
           agentLoopContext?.runContext,
           "agentLoopContext is required to run the run_agent tool"
         );
 
+        const abortSignal = signal ?? null;
+        let childCancellationPromise: Promise<void> | null = null;
+        const finalizeAndReturn = async <T>(
+          result: Result<T, MCPError>
+        ): Promise<Result<T, MCPError>> => {
+          await (childCancellationPromise ?? Promise.resolve());
```

Comment:
nit: I like this better:
```suggestion
          if (childCancellationPromise) {
            await childCancellationPromise;
          }
```